components/vanity/VanityForm.tsx

```diff
@@ -58,15 +61,14 @@ export default function VanityForm() {
               await gqlQueries.createVanityLink({
                 data: vals,
               });
-              
-            } catch (error) {
-
+              setFormData(vals);
+              setVanitySuccess(true);
+            } catch ( error ) {
               setVanitySuccess(false);
-              console.error(error);
-              
+              console.error(  error );
+              setErrMsg( (error as any).response.errors[0].message  )
```

Comment:
Make sure that the value passed into `setErrMsg` is always string. Feel free to resolve this comment if you're sure that the value inside is always going to be a string.

Reply:
im pretty sure it should be a string, but ill add some protection just in caase